core/src/test/java/com/datastax/oss/driver/internal/core/metadata/DefaultEndPointTest.java

```diff
@@ -41,7 +45,32 @@ public void should_create_from_literal_ipv4_address() {
   @Test
   public void should_create_from_literal_ipv6_address() {
     DefaultEndPoint endPoint = new DefaultEndPoint(new InetSocketAddress("::1", 9042));
-    assertThat(endPoint.asMetricPrefix()).isEqualTo("0:0:0:0:0:0:0:1:9042");
+    assertThat(endPoint.asMetricPrefix()).isEqualTo("0:0:0:0:0:0:0:1:localhost:9042");
+  }
+
+  @Test
+  public void should_create_from_ipv4_exact_type_address() throws UnknownHostException {
+    // given
+    byte[] ipAddr = new byte[] {127, 0, 0, 1};
+    InetAddress inet4Address = Inet4Address.getByAddress("localhost", ipAddr);
+    // when
+    DefaultEndPoint defaultEndPoint =
+        new DefaultEndPoint(new InetSocketAddress(inet4Address, 9042));
+    // then
+    assertThat(defaultEndPoint.asMetricPrefix()).isEqualTo("localhost:9042");
+  }
+
+  @Test
+  public void should_create_from_ipv6_exact_type_address() throws UnknownHostException {
+    // given
+    byte[] ipAddr = new byte[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16};
+    InetAddress inet4Address = Inet6Address.getByAddress("localhost", ipAddr);
+    // when
+    DefaultEndPoint defaultEndPoint =
+        new DefaultEndPoint(new InetSocketAddress(inet4Address, 9042));
+    // then
+    assertThat(defaultEndPoint.asMetricPrefix())
+        .isEqualTo("localhost:102:304:506:708:90a:b0c:d0e:f10:9042");
```

Comment:
without this change, this was evaluating to `locahost:9042` as well, causing `llegalArgumentException` when trying to register duplicated metric name in the `MetricRegistry`:
```
 public <T extends Metric> T register(String name, T metric) throws IllegalArgumentException {
        if (metric instanceof MetricSet) {
            registerAll(name, (MetricSet) metric);
        } else {
            final Metric existing = metrics.putIfAbsent(name, metric);
            if (existing == null) {
                onMetricAdded(name, metric);
            } else {
                throw new IllegalArgumentException("A metric named " + name + " already exists");
            }
        }
        return metric;
    }
```
- This change is covering only the case when one IP address is resolving to two: ipv4 and ipv6. I am not sure if it is possible that one address will resolve to more addresses with the same IP - I think it is not.
- We have the same problem in the `SniEndPoint#SniEndPoint` but I think that this problem should not occur there.
let me know wdyt